modules/core/shared/src/main/scala/lucuma/core/math/SignalToNoise.scala

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) 2016-2023 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.core.math
+
+import cats.Order
+import cats.Show
+import eu.timepit.refined.types.numeric.PosBigDecimal
+import io.circe.Decoder
+import io.circe.Encoder
+import io.circe.JsonNumber
+import lucuma.core.optics.Format
+import monocle.Prism
+
+import scala.util.control.Exception.catching
+
+// Signal-to-noise stored as milli-signal to noise Long capped at 9,999,999,999
+// so that it fits in a database numeric(10, 3).
```

Comment:
Why `numeric(10, 3)`?  It seemed "big enough" as far as I can tell without being ridiculously too large.  We can always increase it later if necessary.